Classes/Common/Helper.php

```diff
@@ -893,31 +855,23 @@ public static function whereClause($table, $showHidden = false)
      */
     public static function whereExpression($table, $showHidden = false)
     {
-        $expressionBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
-            ->getQueryBuilderForTable($table)
-            ->expr();
-
         if (\TYPO3_MODE === 'FE') {
             // Should we ignore the record's hidden flag?
-            $ignoreHide = -1;
+            $ignoreHide = 0;
             if ($showHidden) {
                 $ignoreHide = 1;
             }
-
-            // $GLOBALS['TSFE']->sys_page is not always available in frontend.
-            if (is_object($GLOBALS['TSFE']->sys_page)) {
-                $expression = $GLOBALS['TSFE']->sys_page->enableFields($table, $ignoreHide);
-            } else {
-                $pageRepository = GeneralUtility::makeInstance(\TYPO3\CMS\Frontend\Page\PageRepository::class);
-                $GLOBALS['TSFE']->includeTCA();
```

Comment:
There are no more edge cases in which the TCA isn't loaded in FE context. So we don't need this anymore.